### DISCUSSION TITLE
Papers about how PySR actually works?

### DISCUSSION THREAD

**Discussion by andmeier** [MilesCranmer/PySR#199] [Q&A] [answered]
Hi there,

I really like PySR and there seem to be a lot of scientific papers that made use of it. However, I haven‘t found any description or similar that explains how PySR and its Julia backend work respectively. I only read that it uses some regularized evolutionary algorithm, simulated annealing and gradient-free optimization, but no details. Before I start digging into the source, is there any explaination available? Something which makes clear what PySR makes so special and different from other algorithms?

Thanks in advance
Andreas

**Comment by MilesCranmer** [comment]
Will be up on arXiv soonish... I'll paste the link here when it's ready! In summary, it's basically a classical tournament selection genetic algorithm with a few tweaks. The most important tweaks are: BFGS constant optimization (without exact gradients), crossovers, and an adaptive parsimony term.

In terms of the source code, the actual search backend is here: https://github.com/MilesCranmer/SymbolicRegression.jl.

**Comment by MilesCranmer** [comment, marked as answer]
Hey @andmeier,
Sorry for the very long delay. Paper is out; you can find it here: https://github.com/MilesCranmer/pysr_paper and arxiv here: https://arxiv.org/abs/2305.01582.
Cheers,
Miles
